src/inventory/ModelManager.cpp

```diff
@@ -741,46 +759,114 @@ QVariant ModelManager::headerData(int section, Qt::Orientation orientation, int
         return QVariant();
 
     switch (section) {
-        case Column::Name:
-            return tr("Name", "translation model name");
+        case Column::Source:
+            return tr("Source", "Source language for the translation model.");
+        case Column::Target:
+            return tr("Target", "Target language for the translation model.");
+        case Column::Type:
+            return tr("Type", "The model type, affects speed.");
         case Column::Repo:
-            return tr("Repository", "repository from which the translation model originated");
-        case Column::Version:
-            return tr("Version", "translation model version");
+            return tr("Repository", "repository from which the translation model originated.");
+        case Column::LocalVer:
+            return tr("Local Version", "Version of the locally installed model.");
+        case Column::RemoteVer:
+            return tr("Remote Version", "Version of the remote model.");
+        case Column::Installed:
+            return tr("Installed", "Whether the model is available locally, remotely or is out of date.");
         default:
             return QVariant();
     }
 }
 
 QVariant ModelManager::data(const QModelIndex &index, int role) const {
-    if (index.row() >= localModels_.size())
+    if (index.row() >= localModels_.size() + newModels_.size())
         return QVariant();
 
-    Model model = localModels_[index.row()];
+    Model model; // Make sure we have all local models before the remote ones
+    if (index.row() < localModels_.size())
+        model = localModels_[index.row()];
+    else if (index.row() < localModels_.size() + newModels_.size())
+        model = newModels_[index.row() - localModels_.size()];
 
     if (role == Qt::UserRole)
         return QVariant::fromValue(model);
 
     switch (index.column()) {
```

Comment:
There seems to be quite a bit of duplicate code here because we switch on column first, and then on role. Maybe flip that around?

Reply:
Fixd